Usage of validation_alias with AliasChoices with Mypy gives explicit-any error

### Initial Checks

- [x] I confirm that I'm using Pydantic V2

### Description

I've recently tried to bump pydantic version in a project from `2.10.6` to `2.13.4` and it introduced mypy errors related to the usage of an `AliasChoices` as a validation_alias in a  field of a`BaseModel`.

### current mypy config:
```ini
[mypy]
plugins = pydantic.mypy
disallow_any_explicit = true
strict = true

[pydantic-mypy]
init_forbid_extra = true
init_typed = true
warn_required_dynamic_aliases = false
```

### Tests
I've then tested different minor versions to check which one introduced this change, and apparently the next version, `2.11.0a1`, introduced this behavior.

Testing in the example code, we get the following error:
```
test.py:6: error: Explicit "Any" is not allowed  [explicit-any]
```
This seems to indicate that it affects the type checking of the class, and not the typing of the `AliasChoices` itself.

### Questions:
1. Is this is a missing configuration on my part?
1. Is there another way to use validation_alias with multiple options that are properly typed for mypy?

## Related documentation
I've read the documentation about [warn_required_dynamic_aliases](https://pydantic.dev/docs/validation/latest/integrations/dev-tools/mypy/#warn_required_dynamic_aliases) as well which seemed to indicate that the pydantic mypy plugin was properly configured.

### Example Code

```Python
from pydantic import AliasChoices, BaseModel, Field

validation_alias_choices = AliasChoices("choice", "other_choice")


class ModelWithMypyError(BaseModel):
    test_field: str = Field(validation_alias=validation_alias_choices)


class WorkingModel(BaseModel):
    test_field: str = Field(validation_alias="choice")
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.13.4
        pydantic-core version: 2.46.4
          pydantic-core build: profile=release pgo=false
               python version: 3.12.11 (main, Jun 25 2025, 09:57:49) [Clang 16.0.0 (clang-1600.0.26.6)]
                     platform: macOS-26.3.1-arm64-arm-64bit
             related packages: mypy-1.20.2 typing_extensions-4.15.0
                       commit: unknown
```

--- a/docs/integrations/mypy.md
+++ b/docs/integrations/mypy.md
@@ -161,6 +161,8 @@ A configuration file with all plugin strictness flags enabled (and some other my
 
 ### `init_typed`
 
+**Default**: `False`.
+
 Because Pydantic performs [data conversion](../concepts/models.md#data-conversion) by default, the following is still valid at runtime:
 
 ```python {test="skip" lint="skip"}
@@ -171,10 +173,12 @@ class Model(BaseModel):
 Model(a='1')
 ```
 
-For this reason, the plugin will use [`Any`][typing.Any] for field annotations when synthesizing the `__init__` method,
+For this reason, the plugin will use [`Any`][typing.Any] for field annotations when synthesizing the `__init__()` method,
 unless `init_typed` is set or [strict mode](../concepts/strict_mode.md) is enabled on the model.
 
 ### `init_forbid_extra`
+
+**Default**: `False`.
 
 By default, Pydantic allows (and ignores) any extra provided argument:
 
@@ -186,17 +190,14 @@ class Model(BaseModel):
 Model(unrelated=2)
 ```
 
-For this reason, the plugin will add an extra `**kwargs: Any` parameter when synthesizing the `__init__` method, unless
-`init_forbid_extra` is set or the [`extra`][pydantic.ConfigDict.extra] is set to `'forbid'`.
+For this reason, the plugin will add an extra `**kwargs: Any` parameter when synthesizing the `__init__()` method, unless
+`init_forbid_extra` is set or the [`extra`][pydantic.ConfigDict.extra] configuration value is set to `'forbid'`.
 
 ### `warn_required_dynamic_aliases`
 
+**Default**: `False`.
+
 Whether to error when using a dynamically-determined alias or alias generator on a model with
 [`validate_by_name`][pydantic.ConfigDict.validate_by_name] set to `False`. If such aliases are
-present, mypy cannot properly type check calls to `__init__`. In this case, it will default to
-treating all arguments as not required.
-
-!!! note "Compatibility with `Any` being disallowed"
-    Some mypy configuration options (such as [`disallow_any_explicit`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_any_explicit))
-    will error because the synthesized `__init__` method contains [`Any`][typing.Any] annotations. To circumvent the issue, you will have
-    to enable both `init_forbid_extra` and `init_typed`.
+present, mypy cannot properly type check calls to `__init__()`. In this case, it will default to
+treating all arguments as not required, by synthesizing the `__init__()` as `(**kwargs: Any) -> None`.

--- a/pydantic/mypy.py
+++ b/pydantic/mypy.py
@@ -345,7 +345,7 @@ class PydanticModelField:
         if typed or strict:
             type_annotation = self.expand_type(current_info, api, include_root_type=True)
         else:
-            type_annotation = AnyType(TypeOfAny.explicit)
+            type_annotation = AnyType(TypeOfAny.special_form)
 
         return Argument(
             variable=variable,
@@ -952,7 +952,7 @@ class PydanticModelTransformer:
 
         if not self.should_init_forbid_extra(fields, config):
             var = Var('kwargs')
-            args.append(Argument(var, AnyType(TypeOfAny.explicit), None, ARG_STAR2))
+            args.append(Argument(var, AnyType(TypeOfAny.special_form), None, ARG_STAR2))
 
         add_method(self._api, self._cls, '__init__', args=args, return_type=NoneType())
 
@@ -983,7 +983,7 @@ class PydanticModelTransformer:
             )
         if not self.should_init_forbid_extra(fields, config):
             var = Var('kwargs')
-            args.append(Argument(var, AnyType(TypeOfAny.explicit), None, ARG_STAR2))
+            args.append(Argument(var, AnyType(TypeOfAny.special_form), None, ARG_STAR2))
 
         args = args + [fields_set_argument] if is_root_model else [fields_set_argument] + args
 

--- a/tests/mypy/modules/plugin_no_any_dynamic_alias.py
+++ b/tests/mypy/modules/plugin_no_any_dynamic_alias.py
@@ -1,0 +1,7 @@
+"""Asserts that no explicit Any error is raised when using dynamic aliases (which results in constructors being synthesized as `(**kwargs: Any) -> None`)."""
+
+from pydantic import AliasChoices, BaseModel, Field
+
+
+class ModelWithAliasChoices(BaseModel):
+    test_field: str = Field(default='', validation_alias=AliasChoices('choice', 'other_choice'))

--- a/tests/mypy/outputs/mypy-plugin-strict-no-any_ini/plugin_no_any_dynamic_alias.py
+++ b/tests/mypy/outputs/mypy-plugin-strict-no-any_ini/plugin_no_any_dynamic_alias.py
@@ -1,0 +1,8 @@
+"""Asserts that no explicit Any error is raised when using dynamic aliases (which results in constructors being synthesized as `(**kwargs: Any) -> None`)."""
+
+from pydantic import AliasChoices, BaseModel, Field
+
+
+class ModelWithAliasChoices(BaseModel):
+    test_field: str = Field(default='', validation_alias=AliasChoices('choice', 'other_choice'))
+# MYPY: error: Required dynamic aliases disallowed  [pydantic-alias]

--- a/tests/mypy/test_mypy.py
+++ b/tests/mypy/test_mypy.py
@@ -111,6 +111,7 @@ cases: list[ParameterSet | tuple[str, str]] = [
         ('mypy-plugin.ini', 'final_with_default.py'),
         ('mypy-plugin.ini', 'create_model_var.py'),
         ('mypy-plugin-strict-no-any.ini', 'dataclass_no_any.py'),
+        ('mypy-plugin-strict-no-any.ini', 'plugin_no_any_dynamic_alias.py'),
         ('mypy-plugin-very-strict.ini', 'metaclass_args.py'),
         ('pyproject-plugin-no-strict-optional.toml', 'no_strict_optional.py'),
         ('pyproject-plugin-strict-equality.toml', 'strict_equality.py'),

